pkg/client/interfaces_test.go

```diff
@@ -0,0 +1,222 @@
+/*
+Copyright the Sonobuoy contributors 2019
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package client
+
+import (
+	"fmt"
+	"testing"
+)
+
+type ConfigTestCase struct {
+	config        ConfigValidator
+	valid         bool
+	expectedError string
+}
+
+func checkConfigIsValid(tc ConfigTestCase) error {
```

Comment:
I'd make this entire file one table driven test. You already defined the test case type and the "check" function, but instead I'd define those in the test function itself to prevent namespace pollution.

Then like the other table driven test I mentioned, add a `desc` field and run them all as subtests.

So it would be something like:

```
testCases := []testCase{
  {desc: "valid deleteConfig" ...},
  {desc: "invalid deleteConfig because of xyz" ...},
  {desc: "valid fooConfig" ...},
  {desc: "invalid fooConfig" ...},
t.Run(t.desc,...{

}
```

All the test cases already share the type, interface, etc. By breaking it into completely separate tests you're just making it unclear that you're really testing the same functionality in every case.

Reply:
This structure was a side effect of these being separate tests to begin with. I extracted the checking logic, but didn't go the extra step to merge into one test function. I'll update it.

Reply:
👍 I probably should have seen that; I thought they were brand new tests. Thanks for combining them.